src/ngraph/frontend/onnx_import/utils/broadcasting.cpp

```diff
@@ -54,53 +55,95 @@ static std::vector<ngraph::Shape> calculate_numpy_broadcast_shape(ngraph::Shape
     return {output_shape, left_shape, right_shape};
 }
 
+/// \brief      Broadcast input node.
+///
+/// \note       The source shape do not have to be the actual shape of input node. This implies we
+///             may expand input node.
+///
+/// \param[in]  node          The input Node to be broadcasted.
+/// \param[in]  output_shape  The output shape.
+/// \param[in]  source_shape  The source shape from which we want to broadcast input node.
+///
+/// \return     The boroadcasted Node.
+///
+static std::shared_ptr<ngraph::Node> broadcast(const std::shared_ptr<ngraph::Node>& node,
+                                               const ngraph::Shape& output_shape,
+                                               const ngraph::Shape& source_shape)
```

Comment:
If possible, make `source_shape` optional and default to `node->get_shape()`

Reply:
Unfortunately at this moment this would be unused function and this causes `clang` error: `error: unused function 'broadcast' [-Werror,-Wunused-function]`